scripts/generate-changelog.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2017 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Generates entries for CHANGELOG.md from pull requests
+ *
+ * Reads changelog entries from pull requests merged since the last release tag.
+ * Changelog entries are lines within the first PR comment that matches /^\[[a-z\-]+\]/ like `[new-rule]` and `[bugfix]`
+ */
+
+// tslint:disable:no-console
+
+import GitHubApi = require("github");
+import * as fs from "fs";
+
+const github = new GitHubApi({
+    host: "api.github.com",
+    protocol: "https",
+    timeout: 5000,
+});
+
+const repoInfo = {
+    owner: "palantir",
+    repo: "tslint",
+};
+
+// authenticate
+const auth: GitHubApi.Auth = {
+    token: fs.readFileSync("../../github_token.txt", "utf8").toString().trim(),
```

Comment:
Environment variable is a standard solution in cooperation with `HISTCONTROL=ignoreboth` (or at least `ignorespace`) in `.bashrc`. And throwing speaking error in case of missing variable